engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -56,25 +57,61 @@ case object NotBlankParameterValidator extends ParameterValidator {
     nodeId
   )
+
   private def isBlankStringLiteral(expression: String): Boolean =
-    BlankStringLiteralPattern.matcher(expression.trim).matches()
+    blankStringLiteralPattern.matcher(expression.trim).matches()
 }
 
 case class FixedValuesValidator(possibleValues: List[FixedExpressionValue]) extends ParameterValidator {
+  override def isValid(paramName: String, value: String, label: Option[String])(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
+    val values = possibleValues.map(possibleValue => possibleValue.expression)
 
-  override def isValid(paramName: String, value: String, label: Option[String])
-                      (implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
+    if (StringUtils.isBlank(value) || values.contains(value))
+      valid(Unit)
+    else
+      invalid(InvalidPropertyFixedValue(paramName, label, value, possibleValues.map(_.expression)))
+  }
+}
 
-    val values = possibleValues.map(possibleValue => possibleValue.expression)
-    if (values.contains(value)) valid(Unit) else invalid(InvalidPropertyFixedValue(paramName, label, value, possibleValues.map(_.expression)))
+case class RegExpParameterValidator(pattern: String, message: String, description: String) extends ParameterValidator {
+  //Blank value should be not validate - we want to chain validators
+  override def isValid(paramName: String, value: String, label: Option[String])(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
+    if (StringUtils.isBlank(value) || Pattern.compile(pattern).matcher(value).matches())
+      valid(Unit)
+    else
+      invalid(MismatchParameter(message, description, paramName, nodeId.id))
   }
 }
 
-case object LiteralIntValidator extends ParameterValidator {
+case object LiteralIntegerValidator extends ParameterValidator {
+  //Blank value should be not validate - we want to chain validators
+  override def isValid(paramName: String, value: String, label: Option[String])(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] =
+    if (StringUtils.isBlank(value) || Try(value.toInt).isSuccess) valid(Unit) else invalid(error(paramName, nodeId.id))
+
+  private def error(paramName: String, nodeId: String): InvalidIntegerLiteralParameter = InvalidIntegerLiteralParameter(
+    "This field value has to be an integer number",
+    "Please fill field by proper integer type",
+    paramName,
+    nodeId
+  )
+}
 
-  override def isValid(paramName: String, value: String, label: Option[String])
-                      (implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
+case object LiteralParameterValidator {
+  lazy val integerValidator: ParameterValidator = LiteralIntegerValidator
 
-    if (Try(value.toInt).isSuccess) valid(Unit) else invalid(ProcessCompilationError.InvalidLiteralIntValue(paramName, label, value))
-  }
+  lazy val numberValidator: RegExpParameterValidator = RegExpParameterValidator(
```

Comment:
Why you've left it here? It shouldn't be in public api. If you want to have it in sample, you should move it there. Also IMO it is not-real example now, because we recommend to use `LiteralIntegerValidator` for ints. Maybe e-mail regexp will be better choice in sample?

Reply:
Number != Integer but Integer == Number.. It's here because we don't want to create many of RegExp validator for number. Creating compiler costs.. So I have create one validator and allow to reuse it. I think it's ok. But we can discuss about it :)